kedro/extras/datasets/api/api_dataset.py

```diff
@@ -98,6 +102,7 @@ def __init__(
             "params": params,
             "headers": headers,
             "auth": auth,
+            "json": json,
```

Comment:
Do we want to validate either `data` or `json` is supplied but not both?

Reply:
Nice suggestion @limdauto . According to the `requests` documentation, they will ignore the `json` parameter if `data` or `files` is passed in.

I'm somewhat in favor of allowing `requests` to handle the edge case, but not entirely opposed to adding the guardrail to make the interaction more explicit; just think it would put the onus of maintaining the guardrail on the kedro community. What do you think?

Reply:
Yes, that sounds good. Thanks for explaining.